tests/TestCase/Database/Expression/IntervalExpressionTest.php

```diff
@@ -0,0 +1,120 @@
+<?php
+declare(strict_types=1);
+
+/**
+ * CakePHP(tm) : Rapid Development Framework (https://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ *
+ * Licensed under The Open Group Test Suite License
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ * @link          https://cakephp.org CakePHP(tm) Project
+ * @since         4.1.0
+ * @license       https://opensource.org/licenses/mit-license.php MIT License
+ */
+namespace Cake\Test\TestCase\Database\Expression;
+
+use Cake\Database\Expression\IntervalExpression;
+use Cake\Database\Query;
+use Cake\Database\Schema\TableSchema;
+use Cake\Database\Type;
+use Cake\Datasource\ConnectionManager;
+use Cake\I18n\FrozenTime;
+use Cake\TestSuite\TestCase;
+
+/**
+ * Tests IntervalExpression class
+ */
+class IntervalExpressionTest extends TestCase
+{
+    /**
+     * @var \Cake\Database\Driver|\PHPUnit\Framework\MockObject\MockObject
+     */
+    protected $driver;
+
+    /**
+     * @var \Cake\Datasource\ConnectionInterface
+     */
+    protected static $connection;
+
+    /**
+     * @var array
+     */
+    protected static $data = [];
+
+    public static function setUpBeforeClass(): void
+    {
+        parent::setUpBeforeClass();
+        self::$connection = ConnectionManager::get('test');
+        self::$data['tz'] = $utc = new \DateTimeZone('UTC');
+        self::$data['interval'] = \DateInterval::createFromDateString('+1 year + 2 seconds + 111 milliseconds');
+        self::$data['date'] = '2021-04-17 02:03:04.321000';
+        self::$data['tmpTable'] = (new TableSchema('interval_test'))
+            ->addColumn('interval_date', ['type' => 'datetimefractional', 'precision' => 6])
+            ->setTemporary(true);
+        foreach (self::$data['tmpTable']->createSql(self::$connection) as $query) {
+            self::$connection->query($query);
+        }
+    }
+
+    /**
+     * @return void
+     */
+    public static function tearDownAfterClass(): void
+    {
+        foreach (self::$data['tmpTable']->dropSql(self::$connection) as $query) {
+            self::$connection->query($query);
+        }
+        self::$connection = self::$data = null;
+        parent::tearDownAfterClass();
+    }
+
+    /**
+     * Tests interval values.
+     *
+     * @return void
+     */
+    public function testInterval()
+    {
+        // Query using direct date value
+        $iExp = new IntervalExpression(
+            new FrozenTime(self::$data['date'], self::$data['tz']),
+            self::$data['interval']
+        );
+        $query = new Query(self::$connection);
```

Comment:
This unit test is difficult to read. Can we follow the format used by other query tests without splitting up every line?

Reply:
I can collapse it into fewer lines. It is split up as I was doing debugging when I needed to see what sql was being produced. Most of the other tests don't actually run queries, however the output of an interval involving a date will be different depending on the server. I found that actually running a query then checking to make sure the result was valid was the best way to test it.